.github/workflows/ci.yml

```diff
@@ -44,6 +44,44 @@ jobs:
       - uses: codecov/codecov-action@v1
         with:
           file: lcov.info
+  test-arrow-types:
+    name: ArrowTypes.jl - Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.3'
+          - '1' # automatically expands to the latest stable 1.x release of Julia
+          - 'nightly'
+        os:
+          - ubuntu-latest
+        arch:
+          - x64
+    defaults:
+      working-directory: src/ArrowTypes
```

Comment:
Including the package name as part of the matrix would eliminate the YAML redundancy but this is the one part where doing that is hard. It may be possible to use symlinks to clean this up

Reply:
In fbbd33c19fcfa522de3cd6b58fbd33694935d85a I've switched over to using symlinks. I'm specifying `working-directory` on each step currently which has the best chance of working but I'll attempt to use `defaults` once I validate GHA works with the symlinks

Reply:
The `working-directory` solution [was a no-go](https://github.com/JuliaData/Arrow.jl/pull/212#issuecomment-868731166). Using `JULIA_PROJECT` now.